docs/tutorials/deployment.md

```diff
@@ -6,64 +6,74 @@ A few basic concepts about this process:
-| C++/Python inference       | ✅          | ✅          | ✅                          |
-+----------------------------+-------------+-------------+-----------------------------+
-| Dynamic resolution         | ✅          | ✅          | ✅                          |
-+----------------------------+-------------+-------------+-----------------------------+
-| Batch size requirement     | Constant    | Dynamic     | Batch inference unsupported |
-+----------------------------+-------------+-------------+-----------------------------+
-| Extra runtime deps         | torchvision | torchvision | Caffe2 ops (usually already |
-|                            |             |             |                             |
-|                            |             |             | included in PyTorch)        |
-+----------------------------+-------------+-------------+-----------------------------+
-| Faster/Mask/Keypoint R-CNN | ✅          | ✅          | ✅                          |
-+----------------------------+-------------+-------------+-----------------------------+
-| RetinaNet                  | ✅          | ✅          | ✅                          |
-+----------------------------+-------------+-------------+-----------------------------+
-| PointRend R-CNN            | ✅          | ❌          | ❌                          |
-+----------------------------+-------------+-------------+-----------------------------+
-| Cascade R-CNN              | ✅          | ❌          | ❌                          |
-+----------------------------+-------------+-------------+-----------------------------+
++----------------------------+-------------+-------------+-------------------+-----------------------------+
+|       Export Method        |   tracing   |  scripting  | tracing/scripting | caffe2_tracing (DEPRECATED) |
++============================+=============+=============+===================+=============================+
+| **Formats**                | TorchScript | TorchScript |       ONNX        | Caffe2, TorchScript, ONNX   |
++----------------------------+-------------+-------------+-------------------+-----------------------------+
+| **Runtime**                | PyTorch     | PyTorch     |    ONNX Runtime   | Caffe2, PyTorch             |
++----------------------------+-------------+-------------+-------------------+-----------------------------+
+| C++/Python inference       | ✅           | ✅           | ✅                 | ✅                           |
++----------------------------+-------------+-------------+-------------------+-----------------------------+
+| Dynamic resolution         | ✅           | ✅           | ✅                 | ✅                           |
++----------------------------+-------------+-------------+-------------------+-----------------------------+
+| Batch size requirement     | Constant    | Dynamic     | Dynamic           | Batch inference unsupported |
```

Comment:
I think Tracing + ONNX will have the same limitation that batch size has to be constant.

Reply:
`torch.onnx.export(input_names=['input0',...],dynamic_axes={'input0': {0: 'batch', ...}})` would replace any numerical value by strings and make these axis dynamic for onnx, for both scripting and tracing

Reply:
I doubt it can automatically "make" the model dynamic - it would only works for simple models. 
The "batch" dimension in the model may not be in any axis: for an intermediate tensor, the "batch" dimension may not exist and should be constructed by a for loop. 
For example, a model that's traced with a bs=2 might have the following IR:
```
tensor = tensor(N, C)
tensor_0 = tensor[0]
tensor_1 = tensor[1]
output0 = operations(tensor_0)  # operations that cannot be batched
output1 = operations(tensor_1)
output = other_operation(output_0, output_1)
```
So it would only work for the same bs=2 in inference. And the "true" dynamic version of this model should contain a for loop over `N`. Is onnx able to automagically translate the above IR into a for loop?

Reply:
Refactored the support table to separate tracing and scripting for ONNX format.